// office-ui-fabric-react/lib-commonjs/components/Breadcrumb/examples/Breadcrumb.Collapsing.Example.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Breadcrumb_1 = require("office-ui-fabric-react/lib-commonjs/Breadcrumb");
var Label_1 = require("office-ui-fabric-react/lib-commonjs/Label");
var labelStyles = {
    root: { margin: '10px 0', selectors: { '&:not(:first-child)': { marginTop: 24 } } },
};
var items = [
    { text: 'Files', key: 'Files', onClick: _onBreadcrumbItemClicked },
    { text: 'This is folder 1', key: 'f1', onClick: _onBreadcrumbItemClicked },
    { text: 'This is folder 2 with a long name', key: 'f2', onClick: _onBreadcrumbItemClicked },
    { text: 'This is folder 3 long', key: 'f3', onClick: _onBreadcrumbItemClicked },
    { text: 'This is non-clickable folder 4', key: 'f4' },
    { text: 'This is folder 5', key: 'f5', onClick: _onBreadcrumbItemClicked, isCurrentItem: true },
];
exports.BreadcrumbCollapsingExample = function () {
    return (React.createElement("div", null,
        React.createElement(Label_1.Label, { styles: labelStyles }, "With no maxDisplayedItems"),
        React.createElement(Breadcrumb_1.Breadcrumb, { items: items, ariaLabel: "Breadcrumb with no maxDisplayedItems", overflowAriaLabel: "More links" }),
        React.createElement(Label_1.Label, { styles: labelStyles }, "With maxDisplayedItems set to 3"),
        React.createElement(Breadcrumb_1.Breadcrumb, { items: items, maxDisplayedItems: 3, ariaLabel: "Breadcrumb with maxDisplayedItems set to 3", overflowAriaLabel: "More links" }),
        React.createElement(Label_1.Label, { styles: labelStyles }, "With maxDisplayedItems set to 2 and overflowIndex set to 1 (second element)"),
        React.createElement(Breadcrumb_1.Breadcrumb, { items: items, maxDisplayedItems: 2, overflowIndex: 1, ariaLabel: "Breadcrumb with maxDisplayedItems set to 2 and overflowIndex set to 1", overflowAriaLabel: "More links" })));
};
function _onBreadcrumbItemClicked(ev, item) {
    console.log("Breadcrumb item with key \"" + item.key + "\" has been clicked.");
}
