// office-ui-fabric-react/lib-amd/components/DocumentCard/examples/DocumentCard.Basic.Example.js
define(["require", "exports", "tslib", "react", "office-ui-fabric-react/lib/DocumentCard", "office-ui-fabric-react/lib/Image", "@uifabric/example-data"], function (require, exports, tslib_1, React, DocumentCard_1, Image_1, example_data_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var previewProps = {
        previewImages: [
            {
                name: 'Revenue stream proposal fiscal year 2016 version02.pptx',
                linkProps: {
                    href: 'http://bing.com',
                    target: '_blank',
                },
                previewImageSrc: example_data_1.TestImages.documentPreview,
                iconSrc: example_data_1.TestImages.iconPpt,
                imageFit: Image_1.ImageFit.cover,
                width: 318,
                height: 196,
            },
        ],
    };
    var DocumentCardActivityPeople = [{ name: 'Annie Lindqvist', profileImageSrc: example_data_1.TestImages.personaFemale }];
    exports.DocumentCardBasicExample = function () { return (React.createElement(DocumentCard_1.DocumentCard, { "aria-label": "Default Document Card with large file name. Created by Annie Lindqvist a few minutes ago.", onClickHref: "http://bing.com" },
        React.createElement(DocumentCard_1.DocumentCardPreview, tslib_1.__assign({}, previewProps)),
        React.createElement(DocumentCard_1.DocumentCardTitle, { title: 'Large_file_name_with_underscores_used_to_separate_all_of_the_words_and_there_are_so_many_words_' +
                'it_needs_truncating.pptx', shouldTruncate: true }),
        React.createElement(DocumentCard_1.DocumentCardActivity, { activity: "Created a few minutes ago", people: DocumentCardActivityPeople }))); };
});
